.github/workflows/test-gcp-dm.yml

```diff
@@ -118,9 +120,121 @@ jobs:
+          echo "GCP_SA_DEPLOYMENT_NAME=sa-cspm-gcp-ci-test-$suffix" >> $GITHUB_ENV
+
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Init Hermit
+        run: ./bin/hermit env -r >> $GITHUB_ENV
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.9"
+
+      - name: Install Poetry
+        run: |
+          curl -sSL https://install.python-poetry.org | python3 -
+          poetry --version
+
+      - id: google-auth
+        name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@v2
+        with:
+          workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }} # this also sets the project name
+          service_account: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+
+      - name: set TF_VAR_stack_version
+        run: |
+          version=$(grep defaultBeatVersion version/version.go | cut -f2 -d "\"")
+          echo "TF_VAR_stack_version=$version" >> $GITHUB_ENV
+
+      - name: Provision Test Environment (EC)
+        id: deploy_ec
+        if: success()
+        working-directory: ${{ env.TEST_ENVS_DIR }}
+        run: |
+          terraform -v
+          terraform init
+          terraform validate
+          terraform apply --auto-approve -target="module.ec_deployment" -target="module.ec_project"
+          terraform output
+          echo "KIBANA_URL=$(terraform output -raw kibana_url)" >> $GITHUB_ENV
+          echo "ES_URL=$(terraform output -raw elasticsearch_url)" >> $GITHUB_ENV
+          echo "ES_USER=$(terraform output -raw elasticsearch_username)" >> $GITHUB_ENV
+
+          export ES_PASSWORD=$(terraform output -raw elasticsearch_password)
+          echo "::add-mask::$ES_PASSWORD"
+          echo "ES_PASSWORD=$ES_PASSWORD" >> $GITHUB_ENV
+
+      - name: Set up GCP Cloud SDK
+        if: always()
+        uses: "google-github-actions/setup-gcloud@v2"
+
+      - name: Deploy GCP Service Account and Agent
+        id: gcp_deploy
+        env:
+          STACK_VERSION: ${{ env.ELK_VERSION }}
+        run: |
+          # Deploys a GCP Service Account
+          cd ${{ env.DEPLOYMENT_MANAGER_DIR }}
+          export DEPLOYMENT_NAME="${{ env.GCP_SA_DEPLOYMENT_NAME }}"
+          export SERVICE_ACCOUNT_NAME="${{ env.GCP_SA_DEPLOYMENT_NAME }}-sa"
+          ./deploy_service_account.sh
+          mv KEY_FILE.json ../../${{ env.INTEGRATIONS_SETUP_DIR }}
+
+          # Installs CSPM GCP integration
+          cd ../../${{ env.INTEGRATIONS_SETUP_DIR }}
+          export SERVICE_ACCOUNT_JSON_PATH="KEY_FILE.json"
+          export DEPLOYMENT_NAME="${{ env.GCP_AGENT_DEPLOYMENT_NAME }}"
+          poetry install
+          poetry run python ./install_cspm_gcp_integration.py
+
+          # Deploys the agent using an existing service account (SERVICE_ACCOUNT_NAME)
+          cd ../../${{ env.DEPLOYMENT_MANAGER_DIR }}
+          . ./set_env.sh && ./deploy.sh
```

Comment:
there's some trickery going on here:
1. we first deploy a service account using the new deployment manager template, which generates a `KEY_FILE.json` with the said service account
2. we then install a CSPM GCP integration that uses `gcp.credentials.json` variable to set the JSON content 
3. then we deploy the agent, but because we've defined a service account name, it won't deploy a service account and instead use the one we deployed earlier

this is all combined into a single step because otherwise we won't be able to share the JSON file